src/Logic/Calculate.js

```diff
@@ -0,0 +1,89 @@
+import operate from './Operate';
+
+export default function calculate(data, buttonName) {
+  let {
+    total, next, operation, answer,
+  } = data;
+  const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  const operators = ['+', 'x', '-', '÷'];
+
+  if (buttonName === '=') {
+    if (total && next && operation) {
+      total = operate(total, next, operation);
+      answer = total;
+      next = null;
+      operation = null;
+      total = null;
+    } else if (answer && next && operation) {
+      total = operate(answer, next, operation);
+      answer = total;
+      next = null;
+      operation = null;
+      total = null;
+    } else if (total && answer && operation) {
+      total = operate(total, answer, operation);
+      answer = total;
+      next = null;
+      operation = null;
+      total = null;
+    }
+  } else if (buttonName === 'AC') {
+    total = null;
+    next = null;
+    operation = null;
+    answer = null;
+  } else if (buttonName === 'A') {
+    total = answer;
+    next = null;
+  } else if (buttonName === 'A' && operation) {
+    next = answer;
+  } else if (buttonName === '+/-') {
+    total *= -1;
+    next *= -1;
+    answer *= -1;
+  } else if (buttonName === '%') {
+    total *= 0.01;
+    next *= 0.01;
```

Comment:
- [ ] This should be handled in the `Operate.js` module. 
screenshot from pathwright 👇🏾 
<img width="921" alt="Screen Shot 2021-02-27 at 5 29 13 PM" src="https://user-images.githubusercontent.com/48804258/109393332-501ea780-7921-11eb-8f51-afa184e22d28.png">
 
<img width="1368" alt="Screen Shot 2021-02-27 at 5 29 26 PM" src="https://user-images.githubusercontent.com/48804258/109393338-5745b580-7921-11eb-8729-990d1096302a.png">

You can do something like 
```
 case '%':
      return one.times(0.01);
```

Reply:
Hi :)
I looked into the % operator on the internet and its math is a little bit complex because of the order of the calculation. For example, if we did 1 + 2 * 4, it suppose multiply 2 and 4 first then sum the result to 1, so it's 9, but on the calculator, it goes 3 then * 4 so it's 12. The correct math is 9. So, let's do it with the %, if we did 2% + 3%, it should print out 0.05 as the requirement said to only make the number a percent-based number, and that's what my calculator does now. Regarding moving the % to the operations function, it will not operate as proper, because it's not a between-numbers operator like + and -, it's a straight one because in this case the second number is considered as 100, so when we do 2 % it's like 2 ÷ 100 so it's 0.02, that's the percentage of 2.

I hope I made myself clear enough, and thanks for the review.